src/lib_ccx/ccx_encoders_spupng.c

```diff
@@ -608,26 +614,71 @@ void black_background(struct pixel_t *target, int target_w, int x, int y, int w,
 // Return 1 on success.
 int spupng_export_string2png(struct spupng_t *sp, uint32_t *str, FILE* output)
 {
+	int error;
 	// Init FreeType if it hasn't been inited yet.
 	if (ft_library==NULL){
-		int error;
+
 		if (error = FT_Init_FreeType(&ft_library))
 		{
 			mprint("\nFailed to init freetype, error code: %d\n", error);
 			return 0;
 		}
-		if (error = FT_New_Face(ft_library, ccx_options.enc_cfg.render_font, 0, &face))
-		{
-			mprint("\n\nFailed to init freetype when trying to init face, error code: %d\n", error);
-			mprint("It's usually caused by the specified font is not found: %s \n", ccx_options.enc_cfg.render_font);
-			mprint("If this is the case, please use -font to manually specify a font that exists. \n\n");
-			return 0;
-		}
-		if (error = FT_Set_Pixel_Sizes(face, 0, FONT_SIZE))
-		{
-			mprint("\nFailed to init freetype when trying to set size, error code: %d\n", error);
-			return 0;
-		}
+	}
+
+	// Init regular font
+	if (error = FT_New_Face(ft_library, ccx_options.enc_cfg.render_font, 0, &face))
+	{
+		mprint("\n\nFailed to init freetype when trying to init face, error code: %d\n", error);
+		mprint("It's usually caused by the specified font is not found: %s \n", ccx_options.enc_cfg.render_font);
+		mprint("If this is the case, please use -font to manually specify a font that exists. \n\n");
+		return 0;
+	}
+	if (error = FT_Set_Pixel_Sizes(face, 0, FONT_SIZE))
+	{
+		mprint("\nFailed to init freetype when trying to set size, error code: %d\n", error);
+		return 0;
+	}
+
+	// Init bold font
+	if (error = FT_New_Face(ft_library, ccx_options.enc_cfg.render_font_bold, 0, &faceb))
+	{
+		mprint("\n\nFailed to init freetype when trying to init face, error code: %d\n", error);
+		mprint("It's usually caused by the specified font is not found: %s \n", ccx_options.enc_cfg.render_font_bold);
+		mprint("If this is the case, please use -font to manually specify a font that exists. \n\n");
+		return 0;
+	}
+	if (error = FT_Set_Pixel_Sizes(faceb, 0, FONT_SIZE))
+	{
+		mprint("\nFailed to init freetype when trying to set size, error code: %d\n", error);
+		return 0;
+	}
+
+	// Init italics font
+	if (error = FT_New_Face(ft_library, ccx_options.enc_cfg.render_font_italics, 0, &facei))
+	{
+		mprint("\n\nFailed to init freetype when trying to init face, error code: %d\n", error);
+		mprint("It's usually caused by the specified font is not found: %s \n", ccx_options.enc_cfg.render_font_italics);
+		mprint("If this is the case, please use -font to manually specify a font that exists. \n\n");
+		return 0;
+	}
+	if (error = FT_Set_Pixel_Sizes(facei, 0, FONT_SIZE))
+	{
+		mprint("\nFailed to init freetype when trying to set size, error code: %d\n", error);
+		return 0;
+	}
+
+	// Init italics and bold font
+	if (error = FT_New_Face(ft_library, ccx_options.enc_cfg.render_font_italics_bold, 0, &facez))
```

Comment:
Repetitive code. It would be better if you use a function instead.

Reply:
Will do.